src/java/htsjdk/samtools/SAMFileReader.java

```diff
@@ -203,6 +204,13 @@ public void close() {
     }
 
     /**
+     * If true, this reader will use asynchronous IO.
+     */
+    public void setUseAsyncIO(final boolean useAsyncIO) {
+        this.useAsyncIO = useAsyncIO;
```

Comment:
this class is deprecated - i don't care about it. We should delete it, see https://github.com/samtools/htsjdk/pull/602

The reason for a different interface is that this is a public class so changing the constructor is breaks API (unless I just add constructors but this class already has a ton of constructors). It's a dead class anyway, let's not worry about it.